### PR TITLE
Print dependence versions

### LINKED ISSUE #438: Gather and print info about dependence libraries version
Is it possible to get a list of the libraries and their version numbers that were used in the binary currently available on kiwix.org?

### PR DESCRIPTION
**Pull request by kelson42**
Fixes #438

Depends on https://github.com/openzim/libzim/pull/656 and https://github.com/kiwix/libkiwix/pull/665

### CODE REVIEW

File: src/version.h [hunk 1]
```diff
@@ -24,9 +24,15 @@
...
+#include <kiwix/version.h>
+#include <zim/version.h>
+
 void version()
 {
-  std::cout << KIWIX_TOOLS_VERSION << std::endl;
+  std::cout << KIWIX_TOOLS_VERSION << std::endl << std::endl;
+  kiwix::printVersions();
+  std::cout << std::endl;
+  zim::printVersions();
```
Comment (veloman-yunkan) [inline src/version.h:35]: Some libs will be duplicated between `kiwix::printVersions()` and `zim::printVersions()`. Maybe it makes sense to print a line before each of them telling what follows. Or the `printVersions()` functions could produce a tree-like output:

```
libkiwix 10.1.2
+ libzim 7.1.0
+ libicu 7.6

libzim 7.1.0
+ libzstd 1.2.3
+ libxapian 4.5.6
+ libicu 7.8.9
```

Comment (kelson42) [reply src/version.h:35]: I agree, I have implemented this readability improvement in kiwix/libkiwix#665.

I have as well created a `getVersionsString()`, this is was not mandatory here, but I need such a function with binaries using Docopt (in particular for a few command line tools in openzim/zim-tools).

If you agree with this two improvements, I plan to make the same i new PR for openzim/libzim.

File: src/version.h [hunk 2]
```diff
@@ -24,9 +24,15 @@
   #define KIWIX_TOOLS_VERSION "undefined"
 #endif
 
+#include <kiwix/version.h>
+#include <zim/version.h>
+
 void version()
 {
-  std::cout << KIWIX_TOOLS_VERSION << std::endl;
+  std::cout << KIWIX_TOOLS_VERSION << std::endl << std::endl;
```
Comment (veloman-yunkan) [inline src/version.h:32]: Doesn't it make sense to start the first line printed by this function with "kiwix-tools "?

Comment (kelson42) [reply src/version.h:32]: The version is the tool version and the kiwix-tools version. It is the same. But believe printing `kiwix-tools x.y.z` to be more confusing than just `x.y.z`.

We could print something like `kiwix-serve x.y.z` (was the case before I believe) but then the code is a bit more complex. But this is doable.

Current compromise seems to me to be the best compromise.

**Review by veloman-yunkan — APPROVED** [review]

File: src/version.h (hunk 2 above)
Comment (kelson42) [reply src/version.h]: @veloman-yunkan Finally changed my mind ;) Thx for the review!

**Pull request merged**